chemprop/cli/train.py

```diff
@@ -824,11 +825,39 @@ def train_model(args, train_loader, val_loader, test_loader, output_dir, scaler,
                     "scale", torch.tensor(scaler.scale_).view(1, -1).float()
                 )
             results = trainer.test(model, test_loader)[0]
-            logger.info(f"Test results: {results}")
+            logger.info(f"Batch Averaged Test results: {results}")
 
             if args.save_preds:
                 predss = trainer.predict(model, test_loader)
                 preds = torch.concat(predss, 0).numpy()
+                if isinstance(test_loader.dataset, MulticomponentDataset):
+                    test_dset = test_loader.dataset.datasets[0]
+                else:
+                    test_dset = test_loader.dataset
+                targets = test_dset.Y
+                mask = torch.from_numpy(np.isfinite(targets))
+                targets = np.nan_to_num(targets, nan=0.0)
+                w_s = torch.from_numpy(test_dset.weights)
+                w_t = model.w_t
+                lt_mask = torch.from_numpy(test_dset.lt_mask) if test_dset.lt_mask[0] else None
```

Comment:
I think you need to check if lt_mask[0] is not None. If it is False and the rest are true, the whole lt mask will be set to None. Same with gt mask.

Reply:
Nice catch. I have made this change.